src/apps/src/eryph-packer/CommandValidationExtensions.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System.CommandLine;
+using Eryph.GenePool.Client.Internal;
+using LanguageExt;
+using LanguageExt.Common;
+
+public static class CommandValidationExtensions
+{
+    public static Argument<T> AddValidation<T,TValidation>(
+        this Argument<T> argument,
+        Func<T, Validation<Error, TValidation>> validator)
+    {
+
+        argument.AddValidator(result =>
+        {
+            var value = result.GetValueOrDefault<T>();
+            var validation = validator(value);
+            var either = validation.ToEither().MapLeft(Error.Many).Map(_ => value);
+            either.IfLeft(l => result.ErrorMessage = $"Invalid argument <{argument.Name}>: {l.Message}");
```

Comment:
In erpyh, I use this method to print errors:
```
    public static string PrintError(Error error) => error switch
    {
        ManyErrors me => string.Join(Environment.NewLine, me.Errors.Map(PrintError)),
        Exceptional ee => ee.ToException().ToString(),
        _ => error.Message
             + error.Inner.Map(i => $"{Environment.NewLine}{PrintError(i)}").IfNone(""),
    };
```
It ensures that the whole error tree is traversed. Also, `Error.Many` will the errors like an array: `[..., ..., ...]`.

Reply:
added this, we really should move this to the functional library...